config/fxdata/objects.cfg

```diff
@@ -713,3 +713,10 @@ Properties = CHOWNED_ON_ROOM_CLAIM
 Name = SPELLBOOK_POSS
 Genre = SPELLBOOK
 Properties = CHOWNED_ON_ROOM_CLAIM
+
+[object136]
```

Comment:
If the custom objects need to go in objects.cfg, is the intention for this to be just linked to custom campaigns? Or will there be some way to link it to specific maps too?

Reply:
We should make some decision "how to pack sprites with specific map"? 
Should we create a `map00201.cfg` + `map00201.tga` or create some `DECLARE_CUSTOM_SPRITE` command or what?
How it should relate with map packs?

Reply:
I think that although some map packs may be fully predefined, like the Deeper Dungeons map pack, most of them should be considered more like 'rule sets' and players can add separate maps inside the existing packs. So following this, the custom images should just be linked to maps, not map packs.

Having a map00201.obj file that defines all objects would work for me. I guess you could define a path to a folder with images there (it stands to reason that when you do custom objects, a single map may have several custom images, so a folder would be good, or otherwise a zip file with all images included).

Having it be done in a script command would be good too, this has the extra bonus of possibly being able to be modified on the fly, so an object changes shape based on level progress. I don't know how difficult it is to realize a script command though.

Reply:
Actually it is a good idea to use minizip library and store maps inside zip files with all their parts.

Reply:
It is easy:tm: difficulty but I think it should have some other script notation because imagine object with 5 different images (view from each side like with creatures). We may restrict to have only one image file but still we have to define 5 different rectangles inside that image file.

Reply:
Related issue:
https://github.com/dkfans/keeperfx/issues/654

Reply:
And what if I want to create a custom trap (i.e. cannon) and play some campaign with it? It is a kind of changing rules.

Reply:
> 
> 
> It is easy™️ difficulty but I think it should have some other script notation because imagine object with 5 different images (view from each side like with creatures). We may restrict to have only one image file but still we have to define 5 different rectangles inside that image file.

If there's a folder (or zip) with files I don't see it being just a single image would be a good idea. Technically it works basically the same, but for users it's a lot harder to understand. I think optimally the script command would refer to a .txt file which has the info about a specific object.

Reply:
> 
> 
> And what if I want to create a custom trap (i.e. cannon) and play some campaign with it? It is a kind of changing rules.

For custom campaigns it would be possible to add a custom objects file if needed.
Or add a new line to the [common] block in the campaign config file that links towards an objects folder, in which case that could be used on mappacks too, but only as an additional possibility.
Or if need be, just have the same script command in every map relevant map linking towards the same image folder.

Reply:
We should declare it more preciesly with all files/map/campaign relations

Reply:
With the current pull it is indeed possible to add a custom object to a custom campaign or mappack, which is good.

Reply:
Now in this branch we have free minizip (as bonus to zlib required for libspng)